-1

Tested at: 5a620d3f2b681e7cb5a845e3b39c6506d6a61fef

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17773/18138/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>In file included from /cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/boost/1.63.0/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/boost/1.63.0/include/boost/lexical_cast/detail/inf_nan.hpp:34:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/boost/1.63.0/include/boost/math/special_functions/sign.hpp:16:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/boost/1.63.0/include/boost/math/tools/config.hpp:388:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/boost/1.63.0/include/boost/detail/fenv.hpp:97:
/cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/fenv.h:58:11: error: no member named 'fenv_t' in the global namespace
  using ::fenv_t;
        ~~^
/cvmfs/cms-ib.cern.ch/nweek-02461/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/fenv.h:59:11: error: no member named 'fexcept_t' in the global namespace
  using ::fexcept_t;
        ~~^
</pre>

--- a/Geometry/HcalCommonData/interface/HcalDDDRecConstants.h
+++ b/Geometry/HcalCommonData/interface/HcalDDDRecConstants.h
@@ -118,7 +118,7 @@ public:
        
 private:
 
-  void                      getOneEtaBin(int subdet, int ieta, int zside,
+  void                      getOneEtaBin(HcalSubdetector subdet, int ieta, int zside,
 					 std::vector<std::pair<int,double>>& phis,
 					 std::map<int,int>& layers, bool planOne,
 					 std::vector<HcalDDDRecConstants::HcalEtaBin>& bins) const;

--- a/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
+++ b/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
@@ -65,9 +65,9 @@ HcalDDDRecConstants::getEtaBins(const int itype) const {
 
   std::vector<HcalDDDRecConstants::HcalEtaBin> bins;
   unsigned int     type     = (itype == 0) ? 0 : 1;
-  int              subdet   = type+1;
+  HcalSubdetector  subdet   = HcalSubdetector(type+1);
   std::vector<int> phiSp;
-  int              subdetSp = hcons.ldMap()->validDet(phiSp);
+  HcalSubdetector  subdetSp = HcalSubdetector(hcons.ldMap()->validDet(phiSp));
   std::map<int,int> layers;
   for (int iz=0; iz<2; ++iz) {
     int zside = 2*iz - 1;
@@ -596,16 +596,17 @@ void HcalDDDRecConstants::specialRBXHBHE(const std::vector<HcalDetId>& idsOld,
 }
 
 
-void HcalDDDRecConstants::getOneEtaBin(int subdet, int ieta, int zside,
+void HcalDDDRecConstants::getOneEtaBin(HcalSubdetector subdet, int ieta, int zside,
 				       std::vector<std::pair<int,double> >& phis,
 				       std::map<int,int>& layers, bool planOne,
 				       std::vector<HcalDDDRecConstants::HcalEtaBin>& bins) const {
 
-  unsigned int lymax = (subdet == static_cast<int>(HcalBarrel)) ? maxLayerHB_+1 : maxLayer_+1;
+  unsigned int lymax = (subdet == HcalBarrel) ? maxLayerHB_+1 : maxLayer_+1;
+  int          type  = (subdet == HcalBarrel) ? 0 : 1;
   double       dphi  = phibin[ieta-1];
   HcalDDDRecConstants::HcalEtaBin etabin = HcalDDDRecConstants::HcalEtaBin(ieta,zside,dphi,etaTable[ieta-1],etaTable[ieta]);
   etabin.phis.insert(etabin.phis.end(),phis.begin(),phis.end());
-  int n = (ieta == iEtaMax[subdet-1]) ? 0 : 1;
+  int n = (ieta == iEtaMax[type]) ? 0 : 1;
   HcalDDDRecConstants::HcalEtaBin etabin0= HcalDDDRecConstants::HcalEtaBin(ieta,zside,dphi,etaTable[ieta-1],etaTable[ieta+n]);
   etabin0.depthStart = hcons.getDepthEta29(phis[0].first,zside,0)+1;
   int dstart = -1;
@@ -613,7 +614,7 @@ void HcalDDDRecConstants::getOneEtaBin(int subdet, int ieta, int zside,
   std::map<int,int>::iterator itr=layers.begin();
   if (layers.size() > 0) {
     int dep = itr->second;
-    if (subdet == static_cast<int>(HcalEndcap) && ieta == iEtaMin[1]) 
+    if (subdet == HcalEndcap && ieta == iEtaMin[type]) 
       dep = hcons.getDepthEta16(subdet,phis[0].first,zside);
     unsigned lymx0 = (layers.size() > lymax) ? lymax : layers.size();
 #ifdef EDM_ML_DEBUG
@@ -636,8 +637,7 @@ void HcalDDDRecConstants::getOneEtaBin(int subdet, int ieta, int zside,
 	} else if (itr->second > dep) {
 	  if (dstart < 0) dstart = dep;
 	  int lmax0 = (lmax >= lmin) ? lmax : lmin;
-	  if (subdet == static_cast<int>(HcalEndcap) && 
-	      ieta+1 == hpar->noff[1] && 
+	  if (subdet == HcalEndcap && ieta+1 == hpar->noff[1] && 
 	      dep > hcons.getDepthEta29(phis[0].first,zside,0)) {
 	    etabin0.layer.push_back(std::pair<int,int>(lmin,lmax0));
 	  } else {
@@ -647,11 +647,10 @@ void HcalDDDRecConstants::getOneEtaBin(int subdet, int ieta, int zside,
 	  lmax = lmin-1;
 	  dep  = itr->second;
 	}
-	if (subdet == static_cast<int>(HcalBarrel)
-	    && ieta == iEtaMax[subdet-1] 
-	    && dep > hcons.getDepthEta16M(1)) break;
-	if (subdet == static_cast<int>(HcalEndcap) && ieta == hpar->noff[1]
-	    && dep > hcons.getDepthEta29M(0,planOne)) {
+	if (subdet == HcalBarrel && ieta == iEtaMax[type] && 
+	    dep > hcons.getDepthEta16M(1)) break;
+	if (subdet == HcalEndcap && ieta == hpar->noff[1] &&
+	    dep > hcons.getDepthEta29M(0,planOne)) {
 	  lmax = lymx0;
 	  break;
 	}
@@ -857,13 +856,11 @@ void HcalDDDRecConstants::initialize(void) {
 
   //Map of special DetId's
   std::vector<int> phis;
-  int              subdet0 = hcons.ldMap()->validDet(phis);
+  HcalSubdetector subdet = HcalSubdetector(hcons.ldMap()->validDet(phis));
   detIdSp_.clear(); detIdSpR_.clear();
-  if ((subdet0 == static_cast<int>(HcalBarrel)) ||
-      (subdet0 == static_cast<int>(HcalEndcap))) {
+  if ((subdet == HcalBarrel) || (subdet == HcalEndcap)) {
     int             phi    = (phis[0] > 0) ? phis[0] : -phis[0];
     int             zside  = (phis[0] > 0) ? 1 : -1;
-    HcalSubdetector subdet = (subdet0 == static_cast<int>(HcalBarrel)) ? HcalBarrel : HcalEndcap;
     int lymax = (subdet == HcalBarrel) ? maxLayerHB_+1 : maxLayer_+1;
     std::pair<int,int>etas = hcons.ldMap()->validEta();
     for (int eta=etas.first; eta<=etas.second; ++eta) {
@@ -890,13 +887,13 @@ void HcalDDDRecConstants::initialize(void) {
       std::pair<int,int> depths = hcons.ldMap()->getDepths(eta);
       for (int ndepth=depths.first; ndepth<=depths.second; ++ndepth) {
 	bool flag = ((subdet == HcalBarrel && eta == iEtaMax[0] && 
-		      ndepth > hcons.getDepthEta16(subdet0,phi,zside)) ||
+		      ndepth > hcons.getDepthEta16(subdet,phi,zside)) ||
 		     (subdet == HcalEndcap && eta == iEtaMin[1] &&
-		      ndepth < hcons.getDepthEta16(subdet0,phi,zside)));
+		      ndepth < hcons.getDepthEta16(subdet,phi,zside)));
 	if (!flag) {
 	  std::vector<int> count(oldDep.size(),0);
-	  int layFront = hcons.ldMap()->getLayerFront(subdet0,eta,phi,zside,ndepth);
-	  int layBack  = hcons.ldMap()->getLayerBack(subdet0,eta,phi,zside,ndepth);
+	  int layFront = hcons.ldMap()->getLayerFront(subdet,eta,phi,zside,ndepth);
+	  int layBack  = hcons.ldMap()->getLayerBack(subdet,eta,phi,zside,ndepth);
 	  for (int lay=layFront; lay<=layBack; ++lay) {
 	    unsigned int l(0);
 	    for (std::map<int,std::pair<int,int> >::iterator itr=oldDep.begin();
@@ -922,9 +919,8 @@ void HcalDDDRecConstants::initialize(void) {
 	  for (unsigned int k=0; k<phis.size(); ++k) {
 	    zside  = (phis[k] > 0) ? 1 : -1;
 	    phi    = (phis[k] > 0) ? phis[k] : -phis[k];
-	    if (subdet == static_cast<int>(HcalEndcap)
-		&& eta == hpar->noff[1] 
-		&& ndepth > hcons.getDepthEta29M(0,true)) break;
+	    if (subdet == HcalEndcap && eta == hpar->noff[1] &&
+		ndepth > hcons.getDepthEta29M(0,true)) break;
 	    HcalDetId newId(subdet,zside*eta,phi,ndepth);
 	    HcalDetId oldId(subdet,zside*eta,phi,odepth);
 	    detIdSp_[newId] = oldId;

